pkg/os/cim/wmi.go

```diff
@@ -254,11 +254,52 @@ func InvokeCimMethod(namespace, class, methodName string, inputParameters map[st
 	return int(result.ReturnValue), outputParameters, nil
 }
 
+// IsNotFound returns true if it's a "not found" error.
+func IsNotFound(err error) bool {
+	return wmierrors.IsNotFound(err)
+}
+
 // IgnoreNotFound returns nil if the error is nil or a "not found" error,
 // otherwise returns the original error.
 func IgnoreNotFound(err error) error {
-	if err == nil || errors.IsNotFound(err) {
+	if err == nil || IsNotFound(err) {
 		return nil
 	}
 	return err
 }
+
+// WithCOMThread runs the given function `fn` on a locked OS thread
+// with COM initialized using COINIT_MULTITHREADED.
+//
+// This is necessary for using COM/OLE APIs directly (e.g., via go-ole),
+// because COM requires that initialization and usage occur on the same thread.
+//
+// It performs the following steps:
+//   - Locks the current goroutine to its OS thread
+//   - Calls ole.CoInitializeEx with COINIT_MULTITHREADED
+//   - Executes the user-provided function
+//   - Uninitializes COM
+//   - Unlocks the thread
+//
+// If COM initialization fails, or if the user's function returns an error,
+// that error is returned by WithCOMThread.
+func WithCOMThread(fn func() error) error {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	if err := ole.CoInitializeEx(0, ole.COINIT_MULTITHREADED); err != nil {
```

Comment:
COM library is initialized for every WMI call, what's the purpose? why not just calling once in driver initialization?